README.md

```diff
@@ -27,6 +27,171 @@ WARNING: If the configuration failed or if the firewall configuration limits
 access to the machine in a bad way, it is most likely be needed to get
 physical access to the machine to fix the issue.
 
+Ansible Facts
+-------------
+
+## Gathering firewall ansible facts
+
+To gather the firewall system role's ansible facts, 
+call the system role with no arguments e.g.
+```yaml
+vars:
+  firewall:
+```
+
+Another option is to gather a more detailed version of the
+ansible facts by using the detailed argument e.g.
+```yaml
+vars:
+  firewall:
+    detailed: true
+```
```

Comment:
Add a flashy warning about th size of the fact when `detailed`?

Reply:
Yes - something like `WARNING: The detailed firewall facts may be approximately 1Mb in size` or whatever the size is.